lnrpc/routerrpc/router_server.go

```diff
@@ -396,3 +412,68 @@ func marshallChannelUpdate(update *lnwire.ChannelUpdate) *ChannelUpdate {
 		FeeRate:         update.FeeRate,
 	}
 }
+
+// TrackPayment picks up a pending payment, waits for the outcome and returns
+// it.
+func (s *Server) TrackPayment(ctx context.Context,
+	request *TrackPaymentRequest) (*PaymentResponse, error) {
+
+	paymentHash, err := lntypes.MakeHash(request.PaymentHash)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("LookupPayment called for payment %v", paymentHash)
+
+	return s.waitForPayment(ctx, paymentHash)
+}
+
+// waitForPayment waits for the result of the payment to be available and
+// returns it as an rpc type.
+func (s *Server) waitForPayment(ctx context.Context, paymentHash lntypes.Hash) (
+	*PaymentResponse, error) {
+
+	// Subscribe to the outcome of this payment.
+	resultChan, err := s.cfg.RouterBackend.Tower.SubscribePayment(
+		paymentHash,
+	)
+	switch err {
```

Comment:
style nit:
```go
switch {
case err == channeldb.ErrPaymentNotInitiated:
     return nil, status.Error(codes.NotFound, err.Error())
case err != nil {
     return nil, err
}
```

Reply:
Okay, nicer. I never felt good about `case nil:`, but this simple improvement didn't occur to me.